mcr_perception_states/ros/src/mcr_perception_states/common/perception_states.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/python
+
+import rospy
+import smach
+import std_msgs.msg
+import mcr_perception_msgs.msg
+
+import random
+
+
+class find_objects(smach.State):
+
+    OBJECT_LIST_TOPIC = '/mcr_perception/object_detector/object_list'
+    EVENT_IN_TOPIC = '/mcr_perception/object_detector/event_in'
+    EVENT_OUT_TOPIC = '/mcr_perception/object_detector/event_out'
+
+    def __init__(self, retries=5):
+        smach.State.__init__(self,
+                             outcomes=['objects_found',
+                                       'no_objects_found'],
+                             input_keys=['found_objects'],
+                             output_keys=['found_objects'])
+        self.object_list_sub = rospy.Subscriber(self.OBJECT_LIST_TOPIC, mcr_perception_msgs.msg.ObjectList,
+                                                self.object_list_cb)
+        self.event_out_sub = rospy.Subscriber(self.EVENT_OUT_TOPIC, std_msgs.msg.String, self.event_out_cb)
+        self.event_in_pub = rospy.Publisher(self.EVENT_IN_TOPIC, std_msgs.msg.String, queue_size=1)
+        self.retries = retries
+
+    def object_list_cb(self, event):
+        self.object_list = event
+
+    def event_out_cb(self, event):
+        self.event_msg = event.data
+
+    def execute(self, userdata):
+        userdata.found_objects = None
+        for i in range(self.retries):
+            self.object_list = None
+            self.event_msg = ""
+
+            rospy.loginfo('Looking for objects (attempt %i/%i)' % (i + 1, self.retries))
+            self.event_in_pub.publish("e_trigger")
+
+            timeout = rospy.Duration.from_sec(10.0)  # wait max of 10.0 seconds
+            start_time = rospy.Time.now()
+
+            while(True):
+                if self.event_msg == "e_done" and self.object_list is not None:
+                    break
+                elif self.event_msg == "e_failed":
+                    rospy.loginfo('Found no objects')
+                    break
+                elif (rospy.Time.now() - start_time) > timeout:
+                    rospy.logerr('Timeout of %f seconds exceeded waiting for object_detector' % float(timeout.to_sec()))
+                    break
+                rospy.sleep(0.01)
+
+            if not self.object_list or len(self.object_list.objects) <= 0:
+                rospy.loginfo('Found no objects')
+            else:
+                n = str([obj.name for obj in self.object_list.objects])
+                rospy.loginfo('Found %i objects: %s' % (len(self.object_list.objects), n))
+                break
+
+        if not self.object_list or len(self.object_list.objects) <= 0:
+            rospy.loginfo('No objects in the field of view')
+            return 'no_objects_found'
+
+        userdata.found_objects = self.object_list.objects
+        return 'objects_found'
+
+
+class accumulate_recognized_objects_list(smach.State):
```

Comment:
This class was added to be used in the industrial robot scenario.
Currently it is not being used .
Can be deleted if its still not being used in domestic .

Reply:
I merged it as it was currently in `mas_common` since I'm not familiar on what is being used in the `mir` repos. Could you open a new pull request with whatever can be removed from the point of view of @b-it-bots/work? @minhnh can review the pull request for @b-it-bots/home .